app/src/pages/inside/projectSettingsPageContainer/content/notifications/ruleGroup/ruleGroup.jsx

```diff
@@ -0,0 +1,330 @@
+import { useIntl } from 'react-intl';
+import CopyIcon from 'common/img/newIcons/copy-inline.svg';
+import PencilIcon from 'common/img/newIcons/pencil-inline.svg';
+import BinIcon from 'common/img/newIcons/bin-inline.svg';
+import AboutIcon from 'common/img/newIcons/icon-about-inline.svg';
+import { PROJECT_SETTINGS_NOTIFICATIONS_EVENTS } from 'analyticsEvents/projectSettingsPageEvents';
+import { showModalAction } from 'controllers/modal';
+import { useTracking } from 'react-tracking';
+import {
+  addProjectNotificationAction,
+  deleteProjectNotificationAction,
+  projectNotificationsStateSelector,
+  updateProjectNotificationAction,
+} from 'controllers/project';
+import { canUpdateSettings } from 'common/utils/permissions';
+import { updateNotificationStateAction } from 'controllers/project/actionCreators';
+import { useDispatch, useSelector } from 'react-redux';
+import { isEmailIntegrationAvailableSelector } from 'controllers/plugins';
+import {
+  activeProjectRoleSelector,
+  activeProjectSelector,
+  userAccountRoleSelector,
+} from 'controllers/user';
+import Parser from 'html-react-parser';
+import PropTypes from 'prop-types';
+import { withTooltip } from 'components/main/tooltips/tooltip';
+import { TextTooltip } from 'components/main/tooltips/textTooltip';
+import { PROJECT_SETTINGS_TAB_PAGE } from 'controllers/pages';
+import { INTEGRATIONS } from 'common/constants/settingsTabs';
+import { LinkComponent } from 'pages/inside/projectSettingsPageContainer/content/notifications/LinkComponent';
+import arrowRightIcon from 'common/img/arrow-right-inline.svg';
+import { DEFAULT_CASE_CONFIG } from '../constants';
+import { convertNotificationCaseForSubmission } from '../utils';
+import {
+  FieldElement,
+  MODAL_ACTION_TYPE_ADD,
+  MODAL_ACTION_TYPE_COPY,
+  MODAL_ACTION_TYPE_EDIT,
+  NotificationRuleContent,
+  RuleList,
+} from '../../elements';
+import { messages } from '../messages';
+import styles from './ruleGroup.scss';
+import { EmptyRuleState } from '../emptyRuleState';
+
+const cx = classNames.bind(styles);
+const COPY_POSTFIX = '_copy';
+
+const RuleItemDisabledTooltip = withTooltip({
+  TooltipComponent: TextTooltip,
+  data: {
+    width: '265px',
+    placement: 'top',
+    tooltipTriggerClass: cx('info-trigger'),
+    dark: true,
+  },
+})(({ children }) => children);
+
+export const RuleGroup = ({ pluginName, typedRules, notifications }) => {
+  const { trackEvent } = useTracking();
+  const { formatMessage } = useIntl();
+
+  const dispatch = useDispatch();
+  const enabled = useSelector(projectNotificationsStateSelector);
+  const projectRole = useSelector(activeProjectRoleSelector);
+  const activeProject = useSelector(activeProjectSelector);
+  const userRole = useSelector(userAccountRoleSelector);
+  const isEmailIntegrationAvailable = useSelector(isEmailIntegrationAvailableSelector);
+
+  const isAvailable = () => canUpdateSettings(userRole, projectRole) && isEmailIntegrationAvailable;
```

Comment:
It's better to use a known value at first. This will not cause additional checking(method calling) in case of false.
`const isAvailable = () => isEmailIntegrationAvailable && canUpdateSettings(userRole, projectRole);`

Reply:
thanks for pointing out